code/parachain/frame/composable-traits/src/assets.rs

```diff
@@ -19,6 +20,76 @@ pub const ASSET_METADATA_SYMBOL_LENGTH: usize = 16;
 pub type BiBoundedAssetName = BiBoundedVec<u8, 1, ASSET_METADATA_NAME_LENGTH>;
 pub type BiBoundedAssetSymbol = BiBoundedVec<u8, 1, ASSET_METADATA_SYMBOL_LENGTH>;
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(JsonSchema, QueryResponses))]
+pub enum ExecuteMsg {
+	// denom is asset u128 to string, for example PICA on Picasso is "1"
+	// all admins default to contract address if not specified
+	#[cfg_attr(feature = "std", returns(CreateResponse))]
+	Create {
+		creation_fee_denom: Option<String>,
+		decimals: Option<u8>,
+		name: Option<String>,
+		symbol: Option<String>,
+		metadata_admin: Option<Addr>,
+		mint_admin: Option<Addr>,
+		burn_admin: Option<Addr>,
+		freeze_admin: Option<Addr>,
+	},
+	// `ed_payment_asset_denom` is used for non sufficient assets in list if any, else ED payed in
+	// PICA
+	#[cfg_attr(feature = "std", returns(String))]
```

Comment:
Why the std split?

Reply:
fixed string return.

because of support https://github.com/CosmWasm/cosmwasm/pull/1483#issuecomment-1581449415